### PR TITLE
refactor: rename lookupName -> packageName

### PR DESCRIPTION
**Pull request by rarkins**
Renames `lookupName` to be `packageName`.

BREAKING CHANGE: Use `packageName` instead of `lookupName` if interacting with Renovate datasources directly.

### CODE REVIEW

File: lib/config/migration.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ export function migrateConfig(
               (item) => item !== 'prEditNotification'
             );
         }
+      } else if (key === 'matchStrings' && is.array(val)) {
+        migratedConfig.matchStrings = val.map((matchString) => {
+          return is.string(matchString)
+            ? matchString.replace(regEx(/<?lookupName>/g), '<?packageName>')
```
Comment (viceice) [inline lib/config/migration.ts:59]:
```suggestion
            ? matchString.replace(regEx(/\\?<lookupName>/g), '?<packageName>')
```

Shouldn't the `?` before? see snapshots above, they look wrong

File: lib/config/migration.ts [hunk 2]
```diff
@@ -441,6 +447,20 @@ export function migrateConfig(
         }
       }
     }
+    if (is.array(migratedConfig.regexManagers)) {
+      const renameMap = {
+        lookupNameTemplate: 'packageNameTemplate',
```
Comment (viceice) [inline lib/config/migration.ts:452]: Why not using the new rename migration?

https://github.com/renovatebot/renovate/blob/00cab7c3a7f9abb09a16bdc34c9681ed3bd04fea/lib/config/migrations/migrations-service.ts#L42-L50

File: lib/config/presets/internal/regex-managers.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export const presets: Record<string, Preset> = {
       {
         fileMatch: ['(^|/|\\.)Dockerfile$', '(^|/)Dockerfile\\.[^/]*$'],
         matchStrings: [
-          '# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>[^\\s]+?)(?: lookupName=(?<lookupName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s(?:ENV|ARG) .+?_VERSION="?(?<currentValue>.+?)"?\\s',
+          '# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s(?:ENV|ARG) .+?_VERSION="?(?<currentValue>.+?)"?\\s',
```
[the remark this answers is not in the document]
Comment (viceice) [reply lib/config/presets/internal/regex-managers.ts:10]: yes, should work.

**Review by viceice — COMMENTED** [review]
first 25 files reviewed

File: lib/modules/manager/dockerfile/extract.spec.ts [hunk 4]
```diff
@@ -219,13 +219,13 @@ describe('modules/manager/dockerfile/extract', () => {
       const res = extractPackageFile('FROM quay.io:1234/node\n').deps;
       expect(res[0]).toMatchInlineSnapshot(`
         Object {
-          "autoReplaceStringTemplate": "{{lookupName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "autoReplaceStringTemplate": "{{packageName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
```
Comment (viceice) [inline lib/modules/manager/dockerfile/extract.spec.ts:222]: We probably need to migrate those too 🤔 As users can references `lookupName` in any templates.

File: lib/modules/manager/pip_requirements/extract.ts [hunk 5]
```diff
@@ -81,15 +81,15 @@ export function extractPackageFile(
...
+        const packageName = gitPackageMatches.groups.gitUrl
           .replace(`${delimiter}${scmPath}`, `:${scmPath}`)
           .replace(`@${currentVersion}`, '');
         dep = {
           ...dep,
           depName,
           currentValue: currentVersion,
           currentVersion: currentVersion,
-          lookupName: lookupName,
+          packageName: packageName,
```
Comment (viceice) [inline lib/modules/manager/pip_requirements/extract.ts:92]:
```suggestion
          packageName,
```

Simplify ?

File: lib/modules/manager/pip_requirements/extract.ts [hunk 6]
```diff
@@ -81,15 +81,15 @@ export function extractPackageFile(
...
         const delimiter = gitPackageMatches.groups.delimiter;
-        const lookupName = gitPackageMatches.groups.gitUrl
+        const packageName = gitPackageMatches.groups.gitUrl
           .replace(`${delimiter}${scmPath}`, `:${scmPath}`)
           .replace(`@${currentVersion}`, '');
         dep = {
           ...dep,
           depName,
           currentValue: currentVersion,
           currentVersion: currentVersion,
```
Comment (viceice) [inline lib/modules/manager/pip_requirements/extract.ts:91]:
```suggestion
          currentValue,
          currentVersion,
```

Simplify

File: lib/modules/manager/terragrunt/modules.ts [hunk 7]
```diff
@@ -35,8 +35,11 @@ export function analyseTerragruntModule(dep: PackageDependency): void {
 
   if (githubRefMatch) {
     dep.depType = 'github';
-    dep.lookupName = githubRefMatch.groups.project.replace(regEx(/\.git$/), '');
-    dep.depName = 'github.com/' + dep.lookupName;
+    dep.packageName = githubRefMatch.groups.project.replace(
+      regEx(/\.git$/),
+      ''
+    );
+    dep.depName = 'github.com/' + dep.packageName;
```
Comment (viceice) [inline lib/modules/manager/terragrunt/modules.ts:42]: We should convert those to string templates soon

File: lib/modules/manager/terragrunt/modules.ts [hunk 8]
```diff
@@ -45,10 +48,10 @@ export function analyseTerragruntModule(dep: PackageDependency): void {
       logger.debug('Terragrunt module contains subdirectory');
       dep.depName = gitTagsRefMatch.groups.path.split('//')[0];
       const tempLookupName = gitTagsRefMatch.groups.url.split('//');
-      dep.lookupName = tempLookupName[0] + '//' + tempLookupName[1];
+      dep.packageName = tempLookupName[0] + '//' + tempLookupName[1];
```
Comment (viceice) [inline lib/modules/manager/terragrunt/modules.ts:51]: same here

File: lib/modules/versioning/pep440/index.spec.ts [hunk 9]
```diff
@@ -84,87 +84,74 @@ describe('modules/versioning/pep440/index', () => {
...
-    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'20.3.0'}  | ${'>=20.3.0,<20.4.0'}
-    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'19.13.1'} | ${'>=19.13.1,<19.14.0'}
-    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'19.13.0'} | ${'>=19.13.0,<19.14.0'}
-    ${'>=19.12.2,<19.13.0'} | ${'auto'}        | ${'19.12.2'}   | ${'19.13.0'} | ${'>=19.13.0,<19.14.0'}
-    ${'>=19.12.2,<20.13.9'} | ${'widen'}       | ${'19.12.2'}   | ${'21.3.1'}  | ${'>=19.12.2,<21.3.2'}
-    ${'>=19.12.2,<19.13.9'} | ${'widen'}       | ${'19.12.2'}   | ${'20.3.1'}  | ${'>=19.12.2,<20.3.2'}
-    ${'>=19.12.2,<19.13.0'} | ${'widen'}       | ${'19.12.2'}   | ${'20.3.1'}  | ${'>=19.12.2,<20.4.0'}
-    ${'>=19.12.2,<19.13.0'} | ${'widen'}       | ${'19.12.2'}   | ${'20.3.0'}  | ${'>=19.12.2,<20.4.0'}
-    ${'>=19.12.2,<19.13.0'} | ${'widen'}       | ${'19.12.2'}   | ${'19.13.1'} | ${'>=19.12.2,<19.14.0'}
-    ${'>=19.12.2,<19.13.0'} | ${'widen'}       | ${'19.12.2'}   | ${'19.13.0'} | ${'>=19.12.2,<19.14.0'}
```
Comment (viceice) [inline lib/modules/versioning/pep440/index.spec.ts:164]: Why all those tests are removed?

File: lib/util/template/index.ts [hunk 10]
```diff
@@ -90,7 +90,7 @@ export const allowedFields = {
   isSingleVersion:
     'true if the upgrade is to a single version rather than a range',
   logJSON: 'ChangeLogResult object for the upgrade',
-  lookupName: 'The full name that was used to look up the dependency.',
+  packageName: 'The full name that was used to look up the dependency.',
```
Comment (viceice) [inline lib/util/template/index.ts:93]: See other comment about config migration.

File: lib/modules/versioning/pep440/index.spec.ts (hunk 9 above)
Comment (rarkins) [reply lib/modules/versioning/pep440/index.spec.ts:164]: Maybe a bad merge

**Review by viceice — APPROVED** [review]

**Pull request merged**